src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/Logging/VisualStudioFeedbackLoggerProvider.InMemoryBuffer.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Text;
+using System.Threading;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor.Logging;
+
+internal partial class VisualStudioFeedbackLoggerProvider
+{
+    /// <summary>
+    /// A circular in memory buffer to store logs in memory. The intent is
+    /// to lazily allocate the buffer as needed, provide a <see cref="ToString"/> method
+    /// for human readability of logs, and prepare the memory for garbage collection
+    /// on call to <see cref="Clear"/>
+    /// </summary>
+    private class InMemoryBuffer(int bufferSize)
```

Comment:
I could use some extra scrutiny and ideas here, maybe from @DustinCampbell ? 

I wanted to make sure the logger wasn't allocating memory for no reason, and cleaned it up when it could. I believe this is a safe way to do it, but if there's a better way I'm definitely keen on learning